syntax = "proto3";
package anime;

service AnimeService {
  rpc FetchAnime (stream Anime) returns (Void) {};
  rpc FetchAnimeRating (stream AnimeRating) returns (Void) {};
}

message Anime {
  uint32 id = 1;
  string name = 2;
  float score = 3;
  string genres = 4;
  string japanese_name = 5;
  string type = 6;
  uint32 episodes = 7;
  string broadcast_start_date = 8;
  string broadcast_end_date = 9;
  string season_premiere = 10;
  string studios = 11;
  string source = 12;
  string age_classification = 13;
  uint32 popularity = 14;
  uint32 watching = 15;
  string synopsis = 16;
}

message AnimeRating {
  uint32 user_id = 1;
  uint32 anime_id = 2;
  uint32 rating = 3;
  WatchingStatus watching_status = 4;
}

message Void {}

enum WatchingStatus {
  UNKNOWN = 0;
  CURRENTLY_WATCHING = 1;
  COMPLETED = 2;
  ON_HOLD = 3;
  DROPPED = 4;
  PLAN_TO_WATCH = 5;
}